Formula/mold.rb

```diff
@@ -70,11 +70,10 @@ def install
       int main(void) { return 0; }
     EOS
 
-    # GCC 12.1.0+ can also use `-fuse-ld=mold`
     linker_flag = case ENV.compiler
-    when :clang then "-fuse-ld=mold"
+    when :clang, /^gcc-1[2-9]$/ then "-fuse-ld=mold"
     when /^gcc(-\d+)?$/ then "-B#{libexec}/mold"
```

Comment:
Swapping can make this more future proof for GCC 20+ (I know that's years away but still):

```suggestion
    when /^gcc(-(\d|10|11))?$/ then "-B#{libexec}/mold"
    when :clang, /^gcc-\d+$/ then "-fuse-ld=mold"
```

Reply:
I did contemplate doing `/^gcc-[1-9][2-9]$/` in order to include GCC `20-99`, so I'm not opposed to this.